go/libraries/utils/autoincr/auto_increment_tracker.go

```diff
@@ -0,0 +1,72 @@
+package autoincr
+
+type AutoIncrementTracker interface {
+	Get(dbname, tableName string) (uint64, bool)
```

Comment:
Rather than Get and Set, just have one method, Reserve. Returns true or false, whether the reservation was successful (false if taken). Then clients try again with the next value as necessary

Instead of this interface being responsible for giving out next values, it just needs to remember what has been given out so integrators can adjust accordingly